cmd/podman/pull.go

```diff
@@ -124,7 +124,12 @@ func pullCmd(c *cli.Context) error {
 		}
 		imgID = newImage[0].ID()
 	} else {
-		newImage, err := runtime.ImageRuntime().New(getContext(), image, c.String("signature-policy"), c.String("authfile"), writer, &dockerRegistryOptions, image2.SigningOptions{}, true, forceSecure)
+		authfile := c.String("authfile")
+		envAuthfile := os.Getenv("REGISTRY_AUTH_FILE")
+		if authfile == "" {
+			authfile = envAuthfile
+		}
+		newImage, err := runtime.ImageRuntime().New(getContext(), image, c.String("signature-policy"), authfile, writer, &dockerRegistryOptions, image2.SigningOptions{}, true, forceSecure)
 		if err != nil {
```

Comment:
Will need to document the environment variable in the man page(s) too, but I know you're planning further PR's or adding to this one, so that can be done later.